_source/_posts/2020-12-11-build-and-secure-an-api-in-python-with-fastapi.md

```diff
@@ -0,0 +1,362 @@
+
+Finally, while FastAPI comes with many of the features you would expect in a REST API framework (like [data validation](https://fastapi.tiangolo.com/features/#validation) and [authentication](https://fastapi.tiangolo.com/features/#security-and-authentication)), it lets you choose your ORM and database of choice. This means that FastAPI can work with your existing data models if you're migrating from an existing Python application.
+
+While FastAPI is an excellent option for building REST APIs in Python, it's not perfect for every situation. If the system you're building relies on Python 3.5 or any earlier Python framework, you won't be able to use FastAPI. FastAPI also isn't a great choice if you're building a server-side web application that needs to display HTML, as it doesn't include all the view helper functions that full-stack frameworks like Django do.
+
+Now that you're familiar with FastAPI at a high level, you're ready to start building your first application. In the rest of this tutorial, you'll see how to get started and secure your endpoints using Okta as your OAuth authorization server.
+
+## Building a New FastAPI Project
+
+In this section, you'll create a new FastAPI project and add a single, unprotected endpoint to your API.
+
+Before you get started, make sure your computer has [Python 3.6+ installed](https://wiki.python.org/moin/BeginnersGuide/Download). FastAPI uses the typing and asynchronous features in Python, so earlier versions of the language won't run it.
+
+### Install FastAPI
+
+Start by creating a new Python project and using pip to [install FastAPI](https://fastapi.tiangolo.com/#installation) and [Uvicorn](https://www.uvicorn.org/) as your asynchronous server:
+
+```bash
+pip install fastapi uvicorn
+pip freeze > requirements.txt
+```
+
+The dependencies will be added to your requirements.txt file.
+
+### Create a New Endpoint
+
+For this tutorial, you'll use a single `main.py` file that contains all your routes. You can [break this file up as your application grows](https://fastapi.tiangolo.com/tutorial/bigger-applications/), but since you'll just be adding a couple of endpoints, you don't need to worry about that now.
+
+Create the `main.py` file in your project's root directory and add the following:
+
+```python
+from fastapi import FastAPI
+
+
+app = FastAPI()
+
+
+# Hello World route
+@app.get("/")
+def read_root():
+    return {"Hello": "World"}
+
+```
+
+### Accessing Your API
+
+When using FastAPI, there are two ways you can access your API. You can use an API explorer like [Postman](https://www.postman.com/), or you can use FastAPI's interactive documentation in your browser. This tutorial will focus on the latter.
+
+To start your application, run the Uvicorn server:
+
+```bash
+uvicorn main:app --reload
+```
+
+Then, open [http://127.0.0.1:8000](http://127.0.0.1:8000) in your browser to see your "Hello, world" endpoint in action:
```

Comment:
```suggestion
Then, open `http://127.0.0.1:8000` in your browser to see your "Hello, world" endpoint in action:
```

We don't typically link to localhost - we use backticks around them instead. It comes from before times when we'd validate all links. It wouldn't be a bad idea to start validating them again, at least weekly. I've heard https://www.drlinkcheck.com/ works well.